<?xml version="1.0"?>
<!--
This launch file registers depth map to rgb image and launches
the node to transfer lane from segmap to pointcloud.
Note: If the nav stack is running with lane plugin, the pointcloud
published by lane_transfer node would be registered on the costmap

Author  : Subramanian Krishnan (subramak@andrew.cmu.edu)
Date    : 19 Sep 2020

Changelog:
    subbu - 9/19 - Initial commit
-->

<launch>

  <!-- Register depth image to rgb image -->
  <node pkg="nodelet" type="nodelet" name="nodelet_manager" args="manager" />

  <node pkg="nodelet" type="nodelet" name="nodelet1" args="load depth_image_proc/register nodelet_manager">
    <remap from="depth/camera_info" to="/frontCamera/depth/camera_info"/>
    <remap from="rgb/camera_info" to="/frontCamera/color/camera_info"/>
    <remap from="depth/image_rect" to="/frontCamera/depth/image_raw"/>
  </node>

  <!-- Launch the image to pointcloud transfer node -->
  <node pkg="autovalet_goal_generation" type="transfer_lane.py" name="lane_transfer"/>

</launch>
